### ISSUE TITLE
Error compiling Vaultwarden with Debian 12

### ISSUE THREAD

**Issue by crsprsp** [tteck/Proxmox#2197]
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

I get this error when trying to install Vaultwarden. 

### Which Linux distribution are you employing?

Debian 12

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

 `  Compiling diesel_migrations v2.1.0
error: could not compile `vaultwarden` (bin "vaultwarden")

Caused by:
  process didn't exit successfully: `/root/.rustup/toolchains/1.74.1-x86_64-unknown-linux-gnu/bin         /rustc --crate-name vaultwarden --edition=2021 src/main.rs --error-format=json --json=diagnostic-         rendered-ansi,artifacts,future-incompat --diagnostic-width=97 --crate-type bin --emit=dep-info,li         nk -C opt-level=3 -C lto=fat -C codegen-units=1 --cfg 'feature="libsqlite3-sys"' --cfg 'feature="         mysql"' --cfg 'feature="postgresql"' --cfg 'feature="sqlite"' -C metadata=5cc24b888ec95882 -C ext         ra-filename=-5cc24b888ec95882 --out-dir /vaultwarden/target/release/deps -C strip=debuginfo -L de         pendency=/vaultwarden/target/release/deps --extern argon2=/vaultwarden/target/release/deps/libarg         on2-f965b5706b85759b.rlib --extern bytes=/vaultwarden/target/release/deps/libbytes-5f8e4317067082         b9.rlib --extern cached=/vaultwarden/target/release/deps/libcached-ab3950892688933f.rlib --extern          chrono=/vaultwarden/target/release/deps/libchrono-ca6fd815a8bbd5be.rlib --extern chrono_tz=/vaul         twarden/target/release/deps/libchrono_tz-99127deb23f10fb0.rlib --extern cookie=/vaultwarden/targe         t/release/deps/libcookie-26344a09aa1c1d52.rlib --extern cookie_store=/vaultwarden/target/release/         deps/libcookie_store-fabd7293801abbae.rlib --extern dashmap=/vaultwarden/target/release/deps/libd         ashmap-c8a99e50e1d2a264.rlib --extern data_encoding=/vaultwarden/target/release/deps/libdata_enco         ding-665d113eb58c7cd8.rlib --extern data_url=/vaultwarden/target/release/deps/libdata_url-02ba273         94ae80a1f.rlib --extern diesel=/vaultwarden/target/release/deps/libdiesel-3a15bea8c9588276.rlib -         -extern diesel_migrations=/vaultwarden/target/release/deps/libdiesel_migrations-b33c747fd726e0d1.         rlib --extern dotenvy=/vaultwarden/target/release/deps/libdotenvy-10925a2b449a8423.rlib --extern          email_address=/vaultwarden/target/release/deps/libemail_address-9ff2bde8b79ac62e.rlib --extern fe         rn=/vaultwarden/target/release/deps/libfern-42e8866fcf4910dd.rlib --extern futures=/vaultwarden/t         arget/release/deps/libfutures-7afec98509d95e92.rlib --extern governor=/vaultwarden/target/release         /deps/libgovernor-2cf88ae223a609d3.rlib --extern handlebars=/vaultwarden/target/release/deps/libh         andlebars-3b56cae0347ac88d.rlib --extern html5gum=/vaultwarden/target/release/deps/libhtml5gum-5f         b34cd94270c085.rlib --extern job_scheduler_ng=/vaultwarden/target/release/deps/libjob_scheduler_n         g-29742a2c40000c3d.rlib --extern jsonwebtoken=/vaultwarden/target/release/deps/libjsonwebtoken-24         da29a453a8b266.rlib --extern lettre=/vaultwarden/target/release/deps/liblettre-bf62e26f5a746f98.r         lib --extern libsqlite3_sys=/vaultwarden/target/release/deps/liblibsqlite3_sys-97583ef66aec8da8.r         lib --extern log=/vaultwarden/target/release/deps/liblog-b28f2804fbe78cd6.rlib --extern num_deriv         e=/vaultwarden/target/release/deps/libnum_derive-b73ef645f5d23a73.so --extern num_traits=/vaultwa         rden/target/release/deps/libnum_traits-220297708c12f124.rlib --extern once_cell=/vaultwarden/targ         et/release/deps/libonce_cell-a62181c1302b112d.rlib --extern openssl=/vaultwarden/target/release/d         eps/libopenssl-131ac04264e33121.rlib --extern paste=/vaultwarden/target/release/deps/libpaste-0ef         eb01a1095ef78.so --extern percent_encoding=/vaultwarden/target/release/deps/libpercent_encoding-7         5b8c6d644ca8e36.rlib --extern pico_args=/vaultwarden/target/release/deps/libpico_args-7281196eded         eca0f.rlib --extern rand=/vaultwarden/target/release/deps/librand-78e94097cab07412.rlib --extern          regex=/vaultwarden/target/release/deps/libregex-c4a876cae9a7cf6c.rlib --extern reqwest=/vaultward         en/target/release/deps/libreqwest-bf889edbd2b090ef.rlib --extern ring=/vaultwarden/target/release         /deps/libring-81dcc05973ad44ed.rlib --extern rmpv=/vaultwarden/target/release/deps/librmpv-911bfd         489e64bbaa.rlib --extern rocket=/vaultwarden/target/release/deps/librocket-186d6c2bf0e13572.rlib          --extern rocket_ws=/vaultwarden/target/release/deps/librocket_ws-80042eeea3fb497d.rlib --extern r         password=/vaultwarden/target/release/deps/librpassword-cc2cb3482e89e44e.rlib --extern semver=/vau         ltwarden/target/release/deps/libsemver-ffc45ed3d357863e.rlib --extern serde=/vaultwarden/target/r         elease/deps/libserde-c738abcf59a91cca.rlib --extern serde_json=/vaultwarden/target/release/deps/l         ibserde_json-bcf029325e3f4477.rlib --extern syslog=/vaultwarden/target/release/deps/libsyslog-b97         3368f8be65baa.rlib --extern time=/vaultwarden/target/release/deps/libtime-847da5f11eb5321e.rlib -         -extern tokio=/vaultwarden/target/release/deps/libtokio-339994af975b5a93.rlib --extern tokio_tung         stenite=/vaultwarden/target/release/deps/libtokio_tungstenite-a5bf766d9e53c6fa.rlib --extern totp         _lite=/vaultwarden/target/release/deps/libtotp_lite-9059eaf2cb566d78.rlib --extern tracing=/vault         warden/target/release/deps/libtracing-1fba4c08f1b7ac31.rlib --extern url=/vaultwarden/target/rele         ase/deps/liburl-04d7fddda399b40f.rlib --extern uuid=/vaultwarden/target/release/deps/libuuid-01ec         c2d2414a0a70.rlib --extern webauthn_rs=/vaultwarden/target/release/deps/libwebauthn_rs-9a64e98d9f         464f75.rlib --extern which=/vaultwarden/target/release/deps/libwhich-b2ec7e20d2a9c2c7.rlib --exte         rn yubico=/vaultwarden/target/release/deps/libyubico-c07d01185afdbf6a.rlib -L native=/vaultwarden         /target/release/build/libsqlite3-sys-6104815f9bb57e90/out -L native=/usr/lib/x86_64-linux-gnu/ -L          native=/usr/lib/x86_64-linux-gnu -L native=/vaultwarden/target/release/build/ring-31a0d97fdd5a8f         0d/out -L native=/vaultwarden/target/release/build/psm-5a4e56e31bbb6701/out --cfg sqlite --cfg my         sql --cfg postgresql` (signal: 9, SIGKILL: kill)
++ error_handler 50 '$STD cargo build --features "sqlite,mysql,postgresql" --release'
++ local exit_code=101
++ local line_number=50
++ local 'command=$STD cargo build --features "sqlite,mysql,postgresql" --release'
++ local 'error_message=\033[01;31m[ERROR]\033[m in line \033[01;31m50\033[m: exit code \033[01;3         1m101\033[m: while executing command \033[33m$STD cargo build --features "sqlite,mysql,postgresql         " --release\033[m'
++ echo -e '\n\033[01;31m[ERROR]\033[m in line \033[01;31m50\033[m: exit code \033[01;31m101\033[         m: while executing command \033[33m$STD cargo build --features "sqlite,mysql,postgresql" --releas         e\033[m'

[ERROR] in line 50: exit code 101: while executing command $STD cargo build --features "sqlite,my         sql,postgresql" --release
++ [[ 50 -eq 24 ]]
+ exit
`

### Please provide detailed steps to reproduce the issue.

Nothing special, just trying to run the install steps.

**Comment by tteck** [comment]
> - Prior to submitting, kindly search the closed issues to check if the problem you are reporting has already been addressed and resolved. If you come across a closed issue that pertains to your problem, please leave a comment on that issue instead of creating a new one. 

https://github.com/tteck/Proxmox/issues/2158

The script runs without any problems from this end. 
```
 _    __            ____ _       __               __         
| |  / /___ ___  __/ / /| |     / /___ __________/ /__  ____ 
| | / / __ `/ / / / / __/ | /| / / __ `/ ___/ __  / _ \/ __ \
| |/ / /_/ / /_/ / / /_ | |/ |/ / /_/ / /  / /_/ /  __/ / / /
|___/\__,_/\__,_/_/\__/ |__/|__/\__,_/_/   \__,_/\___/_/ /_/ 
                                                             
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 109
Using Hostname: vaultwarden
Using Disk Size: 6
Allocated Cores: 4
Allocated RAM: 4096
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Vaultwarden LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 109 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.5 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.114.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Rust
 ✓ Built Vaultwarden 1.30.1
 ✓ Downloaded Web-Vault v2023.12.0
 ✓ Created Service
 ✓ Customized Container
 ✓ Cleaned
 ✓ Set Container to Normal Resources
 ✓ Completed Successfully!

Vaultwarden should be reachable by going to the following URL.
         http://192.168.86.5:8000 
```
![Screenshot 2023-12-16 10 11 38 AM](https://github.com/tteck/Proxmox/assets/21342670/353577a6-dc4c-4a55-b583-3c3f27ff2ab0)

As this issue is unrelated to the script, it is considered closed.

**Comment by tteck** [comment]
If I cannot reproduce the problem on my side, I cannot address an issue that appears to be functioning correctly here.

**Comment by xiccarph** [comment]
I know this is closed, but thought I would add a comment in case it helped someone.
I had a similar problem, with a 0 exit code.
A comment on other compiling error issues led me to something that worked.
I was able to have the script complete when I allocated more memory in the Advanced configuration.
With more memory for the build, in completed.
So not a script problem, just a resource issue.  Not sure why it would be one.
Thanks for your very helpful scripts.
